src/rws_client.cpp

```diff
@@ -147,10 +147,44 @@ RWSClient::RWSResult RWSClient::getMechanicalUnitJointTarget(const std::string&
   return evaluatePOCOResult(httpGet(uri), evaluation_conditions);
 }
 
-RWSClient::RWSResult RWSClient::getMechanicalUnitRobTarget(const std::string& mechunit)
+RWSClient::RWSResult RWSClient::getMechanicalUnitRobTarget(const std::string& mechunit,
+                                                           const Coordinate& coordinate,
+                                                           const std::string& tool,
+                                                           const std::string& wobj)
 {
   std::string uri = generateMechanicalUnitPath(mechunit) + Resources::ROBTARGET;
 
+  std::string args = "";
+  if (!tool.empty())
+  {
+    args += "&tool=" + tool;
+  }
+  if (!wobj.empty())
+  {
+    args += "&wobj=" + wobj;
+  }
+
+  const std::string coordinate_arg = "?coordinate=";
+  switch (coordinate)
+  {
+  case Coordinate::BASE:
```

Comment:
```suggestion
  case BASE:
```
Related to scoped enum.